x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/util/NumericUtils.java

```diff
@@ -132,4 +132,17 @@ public static long unsignedLongMultiplyHigh(long x, long y) {
     private static long twosComplement(long l) {
         return l ^ TWOS_COMPLEMENT_BITMASK;
     }
+
+    /**
+     * Check if the provided double is both finite and a number (i.e. not Double.NaN).
+     * @param dbl The double to verify.
+     * @return The input value.
+     * @throws ArithmeticException if the provided double is either infinite or not a number.
+     */
+    public static double asFiniteNumber(double dbl) {
+        if (Double.isNaN(dbl) || Double.isInfinite(dbl)) {
+            throw new ArithmeticException("not a finite double number: " + dbl);
```

Comment:
@not-napoleon
I've updated `pow()` to use this utility and I've been a bit more verbose than original, since failures here will always be the result of computation, ES not accepting infinites or NaN. Hope that's ok.

Reply:
Sounds good, I like the idea of standardizing these error messages.  I'll keep an eye out for other places we can do so.